daemon/k8s_watcher.go

```diff
@@ -434,27 +441,41 @@ func (d *Daemon) serviceAddFn(obj interface{}) {
 		Namespace: svc.Namespace,
 	}
 
-	clusterIP := net.ParseIP(svc.Spec.ClusterIP)
-	newSI := types.NewK8sServiceInfo(clusterIP)
-
-	// FIXME: Add support for
-	//  - NodePort
+	clusterIPsvc := types.NewK8sServiceInfo(net.ParseIP(svc.Spec.ClusterIP))
 
 	for _, port := range svc.Spec.Ports {
 		p, err := types.NewFEPort(types.L4Type(port.Protocol), uint16(port.Port))
 		if err != nil {
 			log.Errorf("Unable to add service port %v: %s", port, err)
 			continue
 		}
-		if _, ok := newSI.Ports[types.FEPortName(port.Name)]; !ok {
-			newSI.Ports[types.FEPortName(port.Name)] = p
+
+		// Add port to ClusterIP service
+		if _, ok := clusterIPsvc.Ports[types.FEPortName(port.Name)]; !ok {
+			clusterIPsvc.Ports[types.FEPortName(port.Name)] = p
+		}
+
+		// Add NodePort to NodePort services
+		if len(nodePortSvcs) > 0 {
+			if port.NodePort == 0 {
+				log.Warningf("k8s: NodePort service %s with Ports[%s].NodePort == 0",
+					svc.Name, port.Name)
+				continue
+			}
+
+			for i := range nodePortSvcs {
+				if _, ok := nodePortSvcs[i].Ports[types.FEPortName(port.Name)]; !ok {
+					p, _ = types.NewFEPort(types.L4Type(port.Protocol), uint16(port.NodePort))
+					nodePortSvcs[i].Ports[types.FEPortName(port.Name)] = p
```

Comment:
This can' fail as we check it before:
```
                p, err := types.NewFEPort(types.L4Type(port.Protocol), uint16(port.Port))
                if err != nil {
                        log.Errorf("Unable to add service port %v: %s", port, err)
                        continue
                }
```